### PR TITLE
Re-work hls-fetcher to use videojs-contrib-hls components

### PR DESCRIPTION
**Pull request by BrandonOCasey**
use m3u8 parser from hls
use aes-decrpter from hls
re-work to use promises and request module
make decryption optional
any additional playlist from media groups is downloaded

### CODE REVIEW

**Comment by BrandonOCasey** [comment]
build is failing because there are currently no tests (the previous tests were testing decryption and parsing which is now handled externally).

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  - "stable"
+notifications:
+  hipchat:
+    rooms:
+      secure: l5TTd5JuPAW883PtcyaIBcJI9Chr9JpsZPQAEUBKAgIEwzuS6y7t5arlkS1PwH6gi1FADzYDf+OXSIou4GkTSrIetnBcT/SAgF0gBKgIhj+eRkuCfZ4VaC7BPhfZ0hgYRE+5Ejf5BM2MJafRm0pj7OlqG4xKrQZwtuV1te5r3JY=
+  irc:
+    channels:
+      - "chat.freenode.net#videojs"
+    use_notice: true
+# Set up a virtual screen for Firefox.
```
Comment (forbesjo) [inline .travis.yml]: *Chrome

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,29 @@
+sudo: required
+dist: trusty
+language: node_js
+node_js:
+  - "stable"
+notifications:
+  hipchat:
+    rooms:
+      secure: l5TTd5JuPAW883PtcyaIBcJI9Chr9JpsZPQAEUBKAgIEwzuS6y7t5arlkS1PwH6gi1FADzYDf+OXSIou4GkTSrIetnBcT/SAgF0gBKgIhj+eRkuCfZ4VaC7BPhfZ0hgYRE+5Ejf5BM2MJafRm0pj7OlqG4xKrQZwtuV1te5r3JY=
```
Comment (dconnolly) [inline .travis.yml]: Why.

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+      - "chat.freenode.net#videojs"
+    use_notice: true
+# Set up a virtual screen for Firefox.
+before_script:
+  - export CHROME_BIN=/usr/bin/google-chrome
+  - export DISPLAY=:99.0
+  - sh -e /etc/init.d/xvfb start
+env:
+  global:
+  - secure: dM7svnHPPu5IiUMeFWW5zg+iuWNpwt6SSDi3MmVvhSclNMRLesQoRB+7Qq5J/LiKhmjpv1/GlNVV0CTsHMRhZNwQ3fo38eEuTXv99aAflEITXwSEh/VntKViHbGFubn06EnVkJoH6MX3zJ6kbiwc2QdSQbywKzS6l6quUEpWpd0=
```
Comment (dconnolly) [inline .travis.yml]: Why.

File: .travis.yml (hunk 2 above)
Comment (BrandonOCasey) [inline .travis.yml]: all copy pasted from hls, there are not actually any unit tests to run, but it would be nice to have it setup when we do.

**Pull request merged**